apps/api/src/app/account/account.controller.ts

```diff
@@ -154,6 +156,40 @@ export class AccountController {
     }
   }
 
+  @Post('transfer-balance')
+  @UseGuards(AuthGuard('jwt'))
+  public async transferAccountBalance(
+    @Body() { accountIdFrom, accountIdTo, balance }: TransferBalanceDto
+  ) {
+    if (
+      !hasPermission(this.request.user.permissions, permissions.updateAccount)
+    ) {
+      throw new HttpException(
+        getReasonPhrase(StatusCodes.FORBIDDEN),
+        StatusCodes.FORBIDDEN
+      );
+    }
+
```

Comment:
For security reasons, a check is needed that the accounts with id `accountIdFrom` and `accountIdTo` belong to the user.

Can you get the accounts like this:
```
this.accountService.getAccounts(this.request.user.id);
```
And throw a new `NOT_FOUND` exception if the check fails?

Reply:
yes